src/backend_model.cc

```diff
@@ -330,6 +330,22 @@ TritonModel::UpdateModelConfig(
   auto outputs_config = config.mutable_output();
   *outputs_config = updated_config.output();
 
+  if (config.scheduling_choice_case() != updated_config.scheduling_choice_case()) {
+    if (updated_config.has_dynamic_batching()) {
+      auto dynamic_batching_config = config.mutable_dynamic_batching();
+      *dynamic_batching_config = updated_config.dynamic_batching();
+    } else if (updated_config.has_sequence_batching()) {
+      auto sequence_batching_config = config.mutable_sequence_batching();
+      *sequence_batching_config = updated_config.sequence_batching();
+    } else if (updated_config.has_ensemble_scheduling()) {
+      auto ensemble_scheduling_config = config.mutable_ensemble_scheduling();
+      *ensemble_scheduling_config = updated_config.ensemble_scheduling();
+    } else {
+      // updated_config is empty
+      config.clear_scheduling_choice();
+    }
+  }
```

Comment:
The current check seems to be always overwriting the original scheduling choice if the updated config is different? Do we want to have more restriction regarding that? i.e. if the original config specifies sequence / ensemble scheduling, we probably don't want to overwrite it. Perhaps only update the scheduling choice if original config has no scheduling set, or has dynamic batching set?

Reply:
`UpdateModelConfig` is only used by [`TRITONBACKEND_ModelSetConfig`](https://github.com/triton-inference-server/core/blob/1c60840e05eafcc3e0a7fa4d29a996fd7f76f1aa/src/backend_model.cc#L509-L520). `TRITONBACKEND_ModelSetConfig` is only used in the Autocomplete sections of the backends. As we have written the documentation in the header file, the incoming updated config should be overwriting the current config. Otherwise, we are cherry picking updates under certain conditions and that makes it unclear what this function is doing. 

Usage of this method as I see it is, a change to the model configuration happened and the calling code wants to save this new data; this method should do that. Let me know if my understanding is wrong here.

Reply:
> Usage of this method as I see it is, a change to the model configuration happened and the calling code wants to save this new data; this method should do that

I think this should be conditional. Although the TF/ORT backend change has handled it (not filling scheduling option if it has been set), the danger comes from when backend implementer (/ user) tries to have auto-complete support that always set dynamic batching when possible, without realizing that the model may have been configured as a sequence model. In this case the model will end up being served without using sequence batcher.

Reply:
Updated the method to only update the scheduling choice if none is already set. Updated the API in `tritonbackend.h` to reflect this change.